app/routes/issuer.py

```diff
@@ -272,6 +274,151 @@ async def revoke_credential(
     bound_logger.info("Successfully revoked credential.")
 
 
+@router.post("/publish-revocations", status_code=204)
+async def publish_revocations(
+    publish_request: PublishRevocations,
+    auth: AcaPyAuth = Depends(acapy_auth),
+):
+    """
+        Write batch of pending revocations to ledger.
+
+        If no revocation registry id is provided, all pending revocations
+        will be published.
+
+        If no credential revocation id is provided, all pending revocations
+        for the given revocation registry id will be published.
+
+    Parameters:
+    -----------
+        publish_request: Dict[str, List[str]]
+            A map of revocation registry ids to a list of credential revocation ids.
+
+    Returns:
+    --------
+        payload: None
+        status_code: 204
+    """
+    bound_logger = logger.bind(body=publish_request.revocation_registry_credential_map)
+    bound_logger.info("POST request received: Publish revocations")
+
+    async with client_from_auth(auth) as aries_controller:
+        bound_logger.debug("Publishing revocations")
+        await revocation_registry.publish_pending_revocations(
+            controller=aries_controller,
+            rrid2crid=publish_request.revocation_registry_credential_map,
+        )
+
+    bound_logger.info("Successfully published revocations.")
+
+
+@router.post("/clear-pending-revocations", response_model=PublishRevocations)
+async def clear_pending_revocations(
+    purge_pending_request: ClearPendingRevocationsRequest,
+    auth: AcaPyAuth = Depends(acapy_auth),
+):
+    """
+        Clear pending revocations.
+
+        If no revocation registry id is provided, all pending revocations
+        will be cleared.
+
+        If no credential revocation id is provided, all pending revocations
+        for the given revocation registry id will be cleared.
+
+    Parameters:
+    -----------
+        purge_pending_request: Dict[str, List[str]]
+            A map of revocation registry ids to a list of credential revocation ids.
+
+    Returns:
+    --------
+        payload: PublishRevocations
+            The still pending revocations
```

Comment:
Technically it's the revocations that have been cleared? i.e. not still pending

Reply:
No aca-py returns the still pending revocations. If `cred_rev_id` "1", "2" and "3" are pending revocation and you clear `cred_rev_id "2"` then the agent will return `rrid2crid{ rev_reg_id:["1", "3"]}`